GVFS/GVFS.FunctionalTests/Tests/EnlistmentPerFixture/DehydrateTests.cs

```diff
@@ -143,14 +163,363 @@ public void DehydrateShouldFailOnWrongDiskLayoutVersion()
+            this.fileSystem.DeleteDirectory(pathToDelete);
+            GitProcess.Invoke(this.Enlistment.RepoRoot, "commit -a -m \"Delete a directory\"");
+
+            string folderToDehydrate = "TrailingSlashTests";
+            this.DehydrateShouldSucceed(new[] { $"{folderToDehydrate} folder successfully dehydrated." }, confirm: true, noStatus: false, foldersToDehydrate: folderToDehydrate);
+
+            pathToDelete.ShouldNotExistOnDisk(this.fileSystem);
+            GitProcess.Invoke(this.Enlistment.RepoRoot, "checkout HEAD~1");
+            pathToDelete.ShouldBeADirectory(this.fileSystem);
+        }
+
+        [TestCase]
+        public void FolderDehydrateRelativePaths()
+        {
+            string[] foldersToDehydrate = new[]
+            {
+                Path.Combine("..", ".gvfs"),
+                Path.DirectorySeparatorChar + Path.Combine("..", ".gvfs"),
+                Path.Combine("GVFS", "..", "..", ".gvfs"),
+                Path.Combine("GVFS/../../.gvfs"),
+            };
+
+            List<string> errorMessages = new List<string>();
+            foreach (string path in foldersToDehydrate)
+            {
+                errorMessages.Add($"Cannot dehydrate '{path}' folder.  Invalid folder path.");
+            }
+
+            this.DehydrateShouldSucceed(
+                errorMessages.ToArray(),
+                confirm: true,
+                noStatus: false,
+                foldersToDehydrate: foldersToDehydrate);
+        }
+
+        [TestCase]
+        public void FolderDehydrateFolderThatDoesNotExist()
+        {
+            this.DehydrateShouldSucceed(new[] { "DoesNotExist did not exist to dehydrate." }, confirm: true, noStatus: false, foldersToDehydrate: "DoesNotExist");
+        }
+
+        [TestCase]
+        public void FolderDehydrateNewlyCreatedFolderAndFile()
+        {
+            string directoryToCreate = this.Enlistment.GetVirtualPathTo("NewFolder");
+            this.fileSystem.CreateDirectory(directoryToCreate);
+            string fileToCreate = Path.Combine(directoryToCreate, "newfile.txt");
+            this.fileSystem.WriteAllText(fileToCreate, "Test content");
+            GitProcess.Invoke(this.Enlistment.RepoRoot, "add .");
+            GitProcess.Invoke(this.Enlistment.RepoRoot, "commit -m Test");
+
+            this.DehydrateShouldSucceed(new[] { "NewFolder folder successfully dehydrated" }, confirm: true, noStatus: false, foldersToDehydrate: "NewFolder");
+
+            this.Enlistment.UnmountGVFS();
+            fileToCreate.ShouldNotExistOnDisk(this.fileSystem);
+
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                directoryToCreate.ShouldNotExistOnDisk(this.fileSystem);
+            }
+            else
+            {
+                directoryToCreate.ShouldBeADirectory(this.fileSystem);
+            }
+        }
+
+        private void PlaceholderShouldContain(params string[] paths)

```

Comment:
nit: should plurality match between `PlaceholderShouldContain` and `PlaceholdersShouldNotContain`?